src/debugger.ts

```diff
@@ -151,28 +151,31 @@ export async function getDebugConfigurationFromCache(cache: CMakeCache, target:
   }
 
   if (!debuggerPathOverride) {
-    const clang_compiler_regex = /(clang[\+]{0,2})+(?!-cl)/gi;
     // Look for a debugger, in the following order:
-    const cpptoolsExtension = vscode.extensions.getExtension('ms-vscode.cpptools');
-    const cpptoolsDebuggerPath = cpptoolsExtension ? path.join(cpptoolsExtension.extensionPath, "debugAdapters", "lldb-mi", "bin", "lldb-mi") : undefined;
+    // 1. LLDB-MI
+    const clang_compiler_regex = /(clang[\+]{0,2})+(?!-cl)/gi;
     let clang_debugger_path = compiler_path.replace(clang_compiler_regex, 'lldb-mi');
-    // 1. lldb-mi in the compiler path
-    if ((clang_debugger_path.search(new RegExp('lldb-mi')) != -1) && await checkDebugger(clang_debugger_path)) {
-      return createLLDBDebugConfiguration(clang_debugger_path, target);
-    } else if (cpptoolsDebuggerPath && await checkDebugger(cpptoolsDebuggerPath)) {
-      // 2. lldb-mi installed by CppTools
-      return createLLDBDebugConfiguration(cpptoolsDebuggerPath, target);
+    if ((clang_debugger_path.search(new RegExp('lldb-mi')) != -1)) {
+      const cpptoolsExtension = vscode.extensions.getExtension('ms-vscode.cpptools');
+      const cpptoolsDebuggerPath = cpptoolsExtension ? path.join(cpptoolsExtension.extensionPath, "debugAdapters", "lldb-mi", "bin", "lldb-mi") : undefined;
+        // 1a. lldb-mi in the compiler path
+      if (await checkDebugger(clang_debugger_path)) {
+        return createLLDBDebugConfiguration(clang_debugger_path, target);
+      } else if (cpptoolsDebuggerPath && await checkDebugger(cpptoolsDebuggerPath)) {
```

Comment:
It might improve code readability if we remove the `else` keywords in this function.  Each block returns early, so the `else` is not needed.  (not blocking, you can submit without this).

Reply:
Both good ideas and were quick for me to do now.